src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTestTransformer.java

```diff
@@ -56,68 +53,99 @@
   private final TestCaseTransformer testCaseTransformer = new TestCaseTransformer(valueProducer);
   private final StaticLangApiMethodTransformer apiMethodTransformer =
       new StaticLangApiMethodTransformer();
 
-  public JavaGapicSurfaceTestTransformer(GapicCodePathMapper javaPathMapper) {
+  public JavaSurfaceTestTransformer(
+      GapicCodePathMapper javaPathMapper,
+      SurfaceTransformer surfaceTransformer,
+      String unitTestTemplateFile) {
+    this.surfaceTransformer = surfaceTransformer;
     this.pathMapper = javaPathMapper;
+    this.unitTestTemplateFile = unitTestTemplateFile;
   }
 
   @Override
   public List<String> getTemplateFileNames() {
     List<String> fileNames = new ArrayList<>();
-    fileNames.add(UNIT_TEST_TEMPLATE_FILE);
+    fileNames.add(unitTestTemplateFile);
     fileNames.add(SMOKE_TEST_TEMPLATE_FILE);
     fileNames.add(MOCK_SERVICE_IMPL_FILE);
     fileNames.add(MOCK_SERVICE_FILE);
     return fileNames;
   }
 
   @Override
-  public List<ViewModel> transform(Model model, GapicProductConfig productConfig) {
+  public List<ViewModel> transform(ApiModel model, GapicProductConfig productConfig) {
+    SurfaceNamer namer = surfaceTransformer.createSurfaceNamer(productConfig);
+    boolean enableStringFormatFunctions = productConfig.getResourceNameMessageConfigs().isEmpty();
+
     List<ViewModel> views = new ArrayList<>();
-    ProtoApiModel apiModel = new ProtoApiModel(model);
-    for (ProtoInterfaceModel apiInterface : apiModel.getInterfaces()) {
-      GapicInterfaceContext context = createContext(apiInterface, productConfig);
+    for (InterfaceModel apiInterface : model.getInterfaces()) {
+      ImportTypeTable typeTable =
+          surfaceTransformer.createTypeTable(productConfig.getPackageName());
+      InterfaceContext context =
+          surfaceTransformer.createInterfaceContext(
+              apiInterface, productConfig, namer, typeTable, enableStringFormatFunctions);
       views.add(createUnitTestFileView(context));
       if (context.getInterfaceConfig().getSmokeTestConfig() != null) {
-        context = createContext(apiInterface, productConfig);
+        context =
+            surfaceTransformer.createInterfaceContext(
+                apiInterface,
+                productConfig,
+                namer,
+                typeTable.cloneEmpty(),
+                enableStringFormatFunctions);
         views.add(createSmokeTestClassView(context));
       }
     }
-    for (InterfaceModel apiInterface :
-        mockServiceTransformer.getGrpcInterfacesToMock(apiModel, productConfig)) {
-      GapicInterfaceContext context = createContext(apiInterface, productConfig);
-      views.add(createMockServiceImplFileView(context));
 
-      context = createContext(apiInterface, productConfig);
-      views.add(createMockServiceView(context));
+    if (productConfig.getTransportProtocol().equals(TransportProtocol.GRPC)) {
```

Comment:
I don't want to have if-checks in general code for transports unless we know how we are going to refactor later

Reply:
Ok, moved the transport check to inside the MockServiceTransformer functions.

If that's still weird, maybe we could revisit the previous solution that makes MockServiceTransformer an interface and creates a Grpc subclass and Http subclass for that. The problem was the Http subclass just no-op'd because the MockServiceTransformer really is only for grpc solutions.